src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -1236,4 +1249,87 @@ public ParticipationStatus getParticipationStatus( UUID studyId, String particip
 
         return ParticipationStatus.UNKNOWN;
     }
+
+    @Override
+    public ChronicleQuestionnaire getQuestionnaire( UUID studyId, UUID questionnaireEKID ) {
+        SearchApi searchApi;
+        try {
+            ApiClient apiClient = apiClientCache.get( ApiClient.class );
+            searchApi = apiClient.getSearchApi();
+        } catch ( ExecutionException e ) {
+            throw new IllegalStateException( "unable to get apis", e );
+        }
+
+        logger.info( "Retrieving questionnaire: studyId = {}, questionnaire EKID = {}", studyId, questionnaireEKID );
+
+        try {
+            UUID studyEKID = Preconditions.checkNotNull( getStudyEntityKeyId( studyId ), "invalid study: " + studyId );
+
+            // Get questionnaires that neighboring study
+            Map<UUID, List<NeighborEntityDetails>> neighbors = searchApi.executeFilteredEntityNeighborSearch(
+                    studyESID,
+                    new EntityNeighborsFilter(
+                            Set.of( studyEKID ),
+                            java.util.Optional.of( Set.of( questionnaireESID ) ),
+                            java.util.Optional.of( Set.of( studyESID ) ),
+                            java.util.Optional.of( Set.of( partOfESID ) )
+
+                    )
+            );
+            if ( neighbors.containsKey( studyEKID ) ) {
+                ChronicleQuestionnaire questionnaire = new ChronicleQuestionnaire();
+
+                // find questionnaire entity matching given entity key id
+                neighbors.get( studyEKID )
+                        .stream()
+                        .filter( neighbor -> neighbor.getNeighborDetails().isPresent() && neighbor.getNeighborId()
+                                .isPresent() )
+                        .filter( neighbor -> neighbor.getNeighborId().get().toString()
+                                .equals( questionnaireEKID.toString() ) )
+                        .map( neighbor -> neighbor.getNeighborDetails().get() )
+                        .findFirst()
```

Comment:
Since this is meant to be temporary, maybe leave a comment about why you're using `findFirst()`.